src/cs3iface.py

```diff
@@ -9,137 +9,215 @@
     return None
 
-def rmxattr(_endpoint, filename, _ruid, _rgid, key):
-  '''Remove the extended attribute <key> on behalf of the given uid, gid'''
-  try:
-    os.removexattr(_getfilename(filename), key)
-  except (FileNotFoundError, PermissionError) as e:
-    raise IOError(e)
 
-def readfile(_endpoint, filename, _ruid, _rgid):
-  '''Read a file on behalf of the given uid, gid. Note that the function is a generator, managed by Flask.'''
-  log.debug('msg="Invoking readFile" filename="%s"' % filename)
-  try:
-    tstart = time.clock()
-    chunksize = config.getint('io', 'chunksize')
-    f = open(_getfilename(filename), mode='rb', buffering=chunksize)
-    tend = time.clock()
-    log.info('msg="File open for read" filename="%s" elapsedTimems="%.1f"' % (filename, (tend-tstart)*1000))
-    # the actual read is buffered and managed by the Flask server
-    for chunk in iter(lambda: f.read(chunksize), b''):
-      yield chunk
-  except FileNotFoundError as e:
-    # log this case as info to keep the logs cleaner
-    log.info('msg="File not found on read" filename="%s"' % filename)
-    # as this is a generator, we yield the error string instead of the file's contents
-    yield 'ERROR on read'
-    yield 'No such file or directory'
-  except OSError as e:
-    # general case, issue a warning
-    log.warning('msg="Error opening the file for read" filename="%s" error="%s"' % (filename, e))
-    yield 'ERROR on read'
-    yield str(e)
-
-def writefile(_endpoint, filename, ruid, rgid, content, noversion=0):
-  '''Write a file via xroot on behalf of the given uid, gid. The entire content is written
-     and any pre-existing file is deleted (or moved to the previous version if supported).
-     If noversion=1, the write explicitly disables versioning: this is useful for lock files.'''
-  size = len(content)
-  log.debug('msg="Invoking writeFile" filename="%s" size="%d"' % (filename, size))
+def rmxattr(_endpoint, filepath, userid, key):
+  '''Remove the extended attribute <key> using the given userid as access token'''
+  reference = spr.Reference(path=filepath)
+  req = sp.UnsetArbitraryMetadataRequest(ref=reference, arbitrary_metadata_keys=[key])
+  res = ctx['cs3stub'].UnsetArbitraryMetadata(request=req, metadata=[('x-access-token', _authenticate(userid))])
+  if res.status.code != cs3code.CODE_OK:
+    ctx['log'].warning('msg="Failed to rmxattr" filepath="%s" key="%s" exception="%s"' % (filepath, key, res.status.message))
+    raise IOError(res.status.message)
+  ctx['log'].debug('msg="Invoked rmxattr" result="%s"' % res)
+
+
+def readfile(_endpoint, filepath, userid):
+  '''Read a file using the given userid as access token. Note that the function is a generator, managed by Flask.'''
+  tstart = time.clock()
+  # prepare endpoint
+  req = sp.InitiateFileDownloadRequest(ref=spr.Reference(path=filepath))
+  filedownloadres = ctx['cs3stub'].InitiateFileDownload(
+      request=req, metadata=[('x-access-token', _authenticate(userid))])
```

Comment:
Can this be on one line?

Reply:
I guess that was due to linting, if it fits why not (my pylint is rather generous in line width, as you know...)